src/debugger/SteppingManager.js

```diff
@@ -11,50 +11,88 @@
 
 import { BabelNode } from "babel-types";
 import type { DebugChannel } from "./channel/DebugChannel.js";
-import type { StoppedReason } from "./types.js";
+import type { StoppedReason, SteppingType } from "./types.js";
 import invariant from "./../invariant.js";
-import { StepIntoStepper } from "./Stepper.js";
+import { Stepper, StepIntoStepper, StepOverStepper } from "./Stepper.js";
+import type { Realm } from "./../realm.js";
 
 export class SteppingManager {
-  constructor(channel: DebugChannel) {
+  constructor(channel: DebugChannel, realm: Realm, keepOldSteppers?: boolean) {
     this._channel = channel;
+    this._realm = realm;
+    this._steppers = [];
+    this._keepOldSteppers = false;
+    if (keepOldSteppers) this._keepOldSteppers = true;
   }
   _channel: DebugChannel;
-  _stepInto: void | StepIntoStepper;
+  _realm: Realm;
+  _keepOldSteppers: boolean;
+  _steppers: Array<Stepper>;
 
   processStepCommand(kind: "in" | "over" | "out", currentNode: BabelNode) {
     if (kind === "in") {
       this._processStepIn(currentNode);
+    } else if (kind === "over") {
+      this._processStepOver(currentNode);
     }
     // TODO: implement stepOver and stepOut
   }
 
   _processStepIn(ast: BabelNode) {
     invariant(this._stepInto === undefined);
     invariant(ast.loc && ast.loc.source);
-    this._stepInto = new StepIntoStepper(ast.loc.source, ast.loc.start.line, ast.loc.start.column);
+    if (!this._keepOldSteppers) {
+      this._steppers = [];
+    }
+    this._steppers.push(new StepIntoStepper(ast.loc.source, ast.loc.start.line, ast.loc.start.column));
```

Comment:
Create a flow type for {source, line, column} tuple. The code will be much cleaner.